torch_geometric/nn/conv/hgt_conv.py

```diff
@@ -142,29 +148,94 @@ def forward(
-            out_dict[dst_type].append(out)
+        if WITH_PYG_LIB:
+            # parralelize over node-types
+            node_types, xs = x_dict.keys(), x_dict.values()
+
+            # compute K
+            k_wts = [self.k_lin[node_type].weight for node_type in node_types]
+            k_biases = [self.k_lin[node_type].bias for node_type in node_types]
+            k_list = pyg_lib.ops.grouped_matmul(inputs=xs, others=k_wts,
+                                                biases=k_biases)
+            k_dict = {
+                node_type: k_list[i].view(-1, H, D)
+                for i, node_type in enumerate(k_list)
+            }
+
+            # compute Q
+            q_wts = [self.q_lin[node_type].weight for node_type in node_types]
+            q_biases = [self.q_lin[node_type].bias for node_type in node_types]
+            q_list = pyg_lib.ops.grouped_matmul(inputs=xs, others=q_wts,
+                                                biases=q_biases)
+            q_dict = {
+                node_type: q_list[i].view(-1, H, D)
+                for i, node_type in enumerate(q_list)
+            }
+
+            # compute V
+            v_wts = [self.v_lin[node_type].weight for node_type in node_types]
+            v_biases = [self.v_lin[node_type].bias for node_type in node_types]
+            v_list = pyg_lib.ops.grouped_matmul(inputs=xs, others=v_wts,
+                                                biases=v_biases)
+            v_dict = {
+                node_type: v_list[i].view(-1, H, D)
+                for i, node_type in enumerate(v_list)
+            }
+
+            out_dict = {node_type: [] for node_type in node_types}
+
+            # parallelize over edge-types
+            edge_types, e_idxs = edge_index_dict.items(
+            ), edge_index_dict.values()
+            src_types = [edge_type[0] for edge_type in edge_types]
+            dst_types = [edge_type[-1] for edge_type in edge_types]
+            a_rels = [self.a_rel[edge_type] for edge_type in edge_types]
+            m_rels = [self.m_rel[edge_type] for edge_type in edge_types]
+            k_ins = [
+                k_dict[src_type].transpose(0, 1) for src_type in src_types
+            ]
+            k_outs = [
+                k_o_i.transpose(1, 0)
+                for k_o_i in pyg_lib.ops.grouped_matmul(k_ins, a_rels)
+            ]
+            del k_ins
+            v_ins = [
+                v_dict[src_type].transpose(0, 1) for src_type in src_types
+            ]
+            v_outs = [
+                v_o_i.transpose(1, 0)
+                for v_o_i in pyg_lib.ops.grouped_matmul(v_ins, m_rels)
+            ]
+            del v_ins
+            for i, dst_type in enumerate(dst_types):
+                out = self.propagate(e_idxs[i], k=k_outs[i],
+                                     q=q_dict[dst_type], v=v_outs[i],
+                                     rel=self.p_rel[edge_types[i]], size=None)
+                out_dict[dst_type].append(out)
```

Comment:
I think we can only propagate once here if we stack `edge_index_dict` into a single `edge_index` matrix.